packages/react/src/components/TextInput/TextInput-story.js

```diff
@@ -80,7 +129,8 @@ storiesOf('TextInput', module)
     () => (
       <TextInput
         type={select('Form control type (type)', types, 'text')}
-        {...props.TextInputProps()}
+        {...props.textInput()}
+        invalid={props.textInput().readOnly ? false : props.textInput().invalid}
```

Comment:
Nice work ignoring validation here, we definitely want to make sure that's the case in any readonly inputs